speech/grpc/transcribe_streaming.py

```diff
@@ -86,54 +84,48 @@ def _audio_data_generator(buff):
             except queue.Empty:
                 break
 
-        # If `_fill_buffer` adds `None` to the buffer, the audio stream is
-        # closed. Yield the final bit of the buffer and exit the loop.
+        # `None` in the buffer signals that the audio stream is closed. Yield
+        # the final bit of the buffer and exit the loop.
         if None in data:
             stop = True
             data.remove(None)
         yield b''.join(data)
 
 
-def _fill_buffer(audio_stream, buff, chunk, stoprequest):
+def _fill_buffer(buff):
     """Continuously collect data from the audio stream, into the buffer."""
-    try:
-        while not stoprequest.is_set():
-            buff.put(audio_stream.read(chunk))
-    except IOError:
-        pass
-    finally:
-        # Add `None` to the buff, indicating that a stop request is made.
-        # This will signal `_audio_data_generator` to exit.
-        buff.put(None)
+    def callback(in_data, frame_count, time_info, status_flags):
+        buff.put(in_data)
+        return None, pyaudio.paContinue
+    return callback
 
 
 # [START audio_stream]
 @contextlib.contextmanager
-def record_audio(rate, chunk, stoprequest):
+def record_audio(rate, chunk):
     """Opens a recording stream in a context manager."""
+    # Create a thread-safe buffer of audio data
+    buff = queue.Queue()
```

Comment:
I avoided this since `buffer` [is a keyword](https://docs.python.org/2/library/functions.html#buffer)